rotkehlchen/chain/scroll/modules/scroll_airdrop/decoder.py

```diff
@@ -49,11 +58,45 @@ def _decode_airdop_claim(self, context: 'DecoderContext') -> 'DecodingOutput':
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _maybe_enrich_scroll_batch_airdrop(
+            self,
+            context: EnricherContext,
+    ) -> TransferEnrichmentOutput:
+        """Enriches SCR token airdrop claims from Scroll's offchain distributor.
+
+        Handles batch distributions for GitHub/email-based claims that are processed
+        periodically rather than instant on-chain claims.
+
+        See https://scroll-faqs.gitbook.io/faq#23-when-will-i-receive-my-scr-tokens-after-claiming
+        """
+        if (
+            context.transaction.from_address != SCROLL_OFFCHAIN_TOKEN_DISTRIBUTOR and
+            context.transaction.input_data[:4] != EXEC_TRANSACTION_FOUR_BYTES
+        ):
+            return FAILED_ENRICHMENT_OUTPUT
+
+        if (
+            context.event.event_type != HistoryEventType.RECEIVE and
+            context.event.event_subtype != HistoryEventSubType.NONE and
+            context.event.asset != A_SCR
+        ):
+            return FAILED_ENRICHMENT_OUTPUT
+
+        context.event.counterparty = CPT_SCROLL
+        context.event.event_subtype = HistoryEventSubType.AIRDROP
+        context.event.extra_data = {'airdrop_identifier': 'scroll'}
+        context.event.notes = f'Claim {context.event.balance.amount} {context.event.asset.resolve_to_asset_with_symbol().symbol} from scroll airdrop'  # noqa: E501
+
+        return TransferEnrichmentOutput(CPT_SCROLL)
+
     # -- DecoderInterface methods
 
     def addresses_to_decoders(self) -> dict['ChecksumEvmAddress', tuple[Any, ...]]:
         return {SCROLL_TOKEN_DISTRIBUTOR: (self._decode_airdop_claim,)}
 
+    def enricher_rules(self) -> list[Callable]:
+        return [self._maybe_enrich_scroll_batch_airdrop]
+
```

Comment:
this would execute for every transfer... I was thinking that you can have the same checks but do a address to decoder using as address the multisig contract. I believe that would also work and wouldn't trigger for every scroll transfer. You would need to match that there is a transfer of scroll from the multisig 

![imagen](https://github.com/user-attachments/assets/c9a4a016-b821-4e23-94ec-032b27ca878f)